### DISCUSSION TITLE
第1节课笔记(15班)

### DISCUSSION THREAD

**Discussion by ironartisan** [InternLM/Tutorial#36] [笔记]
请在下方提交笔记链接

**Comment by 6forwater29** [comment]
1. 大模型成为发展通用人工智能的重要途径：从专用模型到通用大模型
2. 从模型到应用（数据、预训练、微调、评测、部署、智能体）

**Reply by qwertyuiopasdfghjklzxcn** [reply]
大模型成为发展通用人工智能的重要途径：从专用模型到通用大模型
从模型到应用（数据、预训练、微调、评测、部署、智能体）

**Comment by lelezhou** [comment]
综述性的讲解了大模型的发展：介绍了垂直功能的专用模型如人脸识别到通用能力的通用大模型
介绍了书生模型的开源历程，模型介绍:7b,20b,123b,以及模型的应用（微调，智能体），数据，预训练，评测，部署（LMdeploy）,智能体框架（Lagent）

**Comment by PhD-TianLv** [comment]
[课程PPT.pdf](https://github.com/InternLM/tutorial/files/13826686/default.pdf)

**Comment by 520Enterprise** [comment]
https://www.cnblogs.com/520Enterprise/p/17945399

**Comment by dqyydq** [comment]
https://blog.csdn.net/qq_52410386/article/details/135390249?spm=1001.2014.3001.5502

**Comment by lizhunkg** [comment]
第1节课笔记： https://note.youdao.com/s/PjkZ2Elf

**Comment by IyGnahz** [comment]
1 : https://g6xuom3zbh.feishu.cn/mindnotes/MENgbyDL2mvpTcn8jiHcUsblnqT#mindmap

**Comment by scchy** [comment]
第1节课笔记
https://github.com/scchy/My_Learn/blob/master/LLM_prime/1_InternLM_Introduce.md

**Comment by szhowardhuang** [comment]
大模型成为发展通用人工智能的重要途径，是专用模型到通用模型的一次跨越。
GPT4从语言模态升级到语言，视觉多模态。
书生大模型从几十亿到千亿
问题1： InternLLM 20B性能不错的具体原因可否分析一下？
模型到应用，比如：智能客服，个人助手，行业应用。 以下是评测过程
![image](https://github.com/InternLM/tutorial/assets/1407300/401f3450-b11c-4cdc-a229-ba32d575023d)
书生端到端开源体系如下：
![image](https://github.com/InternLM/tutorial/assets/1407300/24f24428-c45a-47b9-8eec-74dcba73fb9a)
书生数据集：2TB数据包括 50亿文档，2千多万个图像，1千多个视频
问题2：openDataLab开发数据平台的80TB数据和书生数据集之间的关系如何？会重叠吗？另外， 智能标注工具开源吗？
问题3：增量续训是全量参数微调吗？
微调框架XTuner如下图：
![image](https://github.com/InternLM/tutorial/assets/1407300/794f605e-efdb-4224-b3ba-10a61d21f3ba)
问题4：8G显卡微调InternLM 7B是不是只有QLoRA 512才能玩转？
openCompass评测平台架构如下：
![image](https://github.com/InternLM/tutorial/assets/1407300/e1a839f9-ecd6-4c5e-b6c1-e2be5fcd1b5f)
亮点如下：
![image](https://github.com/InternLM/tutorial/assets/1407300/5adf87d4-3e87-4668-8c11-995e771dd6c6)
问题5：LMDeploy技术点：模型并行，Attention优化，计算和缓存优化，Continuous Batching可否详细解释一下？
智能体框图如下：支持ReAct，ReWoo，AutoGPT
![image](https://github.com/InternLM/tutorial/assets/1407300/af29d475-8a50-48fc-abe8-9d42bc78df23)
问题6：AgentLego要做成一个大模型应用开发的开发套件吗？如果只是工具的集合，发展潜力在哪里？或者说如何让开发者有很强的意愿使用此工具箱？
最后，谢谢这个课程go through整个书生开源体系，获得很多有用的信息，也留下不少问题。

**Comment by HarveyLiii** [comment]
本节课的视频介绍了书生·浦语开源项目的全流程，包括数据、预训练、微调、部署、评测和应用。书生·浦语开源体系为大模型的开发和应用提供了全面的支持。

1. 大模型是发展通用人工智能的重要途径，得到了广泛关注。从专用模型向通用模型的发展是人工智能研究的趋势。
2. 书生·浦语开源项目推出了多个开源模型、开源了全链条的工具体系，以及提供了丰富的语料库。
3. 模型微调方面，XTuner能够支持增量续训以及有监督的全参数微调。
4. 评测体系的建立方面，针对大模型的需求提出了自己开源的评测体系OpenCompass，提供了全面的评测能力。
5. 大模型部署所面临各种技术挑战，包括内存开销、动态shape等，提出LMDeploy作为高效的推理框架。
6. 智能体应用方面：针对大模型的局限性和智能体应用的需求，提出Legent智能体框架和Agent Lego工具箱。

**Comment by woodx9** [comment]
打卡：[飞书链接](https://gxloh9mfvxh.feishu.cn/wiki/SkM9wxQBaiyt4NkiAAycPSgOn9b?from=from_copylink)

**Comment by bblgbr** [comment]
大模型的整个流程，上海人工智能研究院都有对应的开源工具：书生·浦语全链路开源体系

- 数据：书生万卷（包含文本，图像，视频等多模态数据，开放平台为OpenDataLab）
- 预训练：InternLM-Train（高可扩展，兼容主流生态）
- 微调：XTuner（7B模型最低8GB显存使用）
- 部署：LMDeploy
- 评测：OpenCompass（6大维度，80+评测集，40万+评测题目）
- 应用：Lagent（智能体框架），AgentLegp（多模态智能体工具箱）

**Comment by yabgdsajghsj** [comment]
[第一课笔记.pdf](https://github.com/InternLM/tutorial/files/13837303/default.pdf)

**Comment by ziptoam** [comment]
第一节课笔记：https://note.youdao.com/s/aKSvVcyS
站在外行人的角度学习大模型构建的过程，了解浦语书生针对这一过程所提供的相关开源工具。

**Comment by cosmowind** [comment]
https://www.cnblogs.com/cosmowind/p/17946828
小白，很多东西都觉得很厉害，但是又记不住。希望接下来能动手操作操作，学习学习。

**Comment by Mrguanglei** [comment]
https://note.youdao.com/s/UtusEjuE

**Comment by pyajagod** [comment]
本节课从数据、预训练、微调、部署、评测和应用部分介绍了书生浦语开源开放体系。
数据：可以用于训练模型；
预训练：可以让模型从无到有；
微调：XTuner能够支持增量续训和监督微调，增量续训可以让模型学到新东西，监督微调可以练习对话
评测：可以通过多种测试对模型进行打分
部署：大模型部署会遇到内存开销、动态shape等挑战。提出LMDeploy作为高效的推理框架。
智能体：Legent智能体框架和AgentLego工具箱可以应对大模型局限性和智能体应用需求

**Comment by lta155** [comment]
http://t.csdnimg.cn/E1jsT

**Comment by motewei** [comment]
https://bwhl75xol6.feishu.cn/docx/NJDbdCEg4oC1UxxN6wQcEPsUnRf?from=from_copylink

**Comment by JakChen797** [comment]
https://juejin.cn/post/7320241554849153076

**Comment by Gaia-yue** [comment]
https://github.com/Gaia-yue/---/blob/main/%E4%B8%80%E6%9C%88/1_introduction.md

**Comment by liuyi6111** [comment]
https://www.mubu.com/doc/5swqx-CyH9G

**Comment by seienn** [comment]
https://gvxxiiwsufy.feishu.cn/wiki/VjZawxkzNiLrXBkNva7cmaC4nvf

**Comment by javey-q** [comment]
- 趋势

专用模型（特定任务，一个模型解决一个问题）

  ->  通用大模型（多任务、多模态）

- 书生 浦语:  InternLM

浦语 训练框架： 

Hybrid Zero ？

- 微调：XTuner

8GB 显存 微调 7B模型

- 评测：OpenCompass

**Comment by wpf-forever** [comment]
https://ywnuqb1iidn.feishu.cn/docx/QUF4dD5ojo43JuxNAuqcQ8kInJg?from=from_copylink

**Comment by Visytudz** [comment]
[书生·浦语全链路体系.pdf](https://github.com/InternLM/tutorial/files/13841665/default.pdf)

**Comment by sokolo05** [comment]
大模型技术在学术界、工业等多方面研究，催生了较多的语言模型，特别是大语言模型的相关研究。AI技术研究从解决单一问题的专用模型—>多任务、多模态通用模型的研究的转化。

书生-浦语大模型技术发展：

轻量级：InternLM-7B—>中量级：InternLM-20B—>重量级轻量级：InternLM-123B

能力：综合考试、知识问题、阅读理解、推理、编程

应用：智能客服、个人助手、行业应用

全链条开源开放体系：

数据—>预训练—>微调—>部署—>评测—>应用

1. 数据：多模态数据融合、精细化处理、价值观对齐
2. 预训练：高可扩展、极致性能优化、兼容主流、开箱即用（配置修改）
3. 微调（XTuner）：增量续训（学习新知识）、有监督微调（全量/部分参数微调）
4. 部署方案技术点：模型并行、低比特量化、attention优化、计算和访存优化、Continuous Batching
   1. 解决方案：LMDeploy—>轻量化、推理、服务
5. 评测（OpenCompass）：6大维度（学科、语言、知识、理解、推理、安全）
6. 应用：
   1. 轻量级智能体框架Lagent：支持多种类型的智能体能力、灵活支持多种大语言模型、简单易拓展，支持丰富工具
   2. 多模态智能体工具箱AgentLego：
      - 丰富的工具集合，尤其是提供了大量视觉、多模态相关领域的前沿算法功能
      - 支持多个主流智能体系统，如 LangChain，Transformers Agent，Lagent 等
      - 灵活的多模态工具调用接口，可以轻松支持各类输入输出格式的工具函数
      - 一键式远程工具部署，轻松使用和调试大模型智能体

**Comment by NANDSS1** [comment]
https://www.yuque.com/nandss/gyyg9p/ak89d59twhxtayh3/edit

**Comment by Guan-Y** [comment]
https://blog.csdn.net/weixin_45932862/article/details/135447313?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135447313%22%2C%22source%22%3A%22weixin_45932862%22%7D

**Comment by g29times** [comment]
https://novagen.notion.site/1-a0759f6caae347d9bf487ca9497754cf?pvs=4
